capture_service/device_mgr.cc

```diff
@@ -96,29 +96,42 @@ absl::StatusOr<std::vector<std::string>> AndroidDevice::ListPackage(PackageListO
         return result.status();
     }
     output = *result;
-
     std::vector<std::string> lines = absl::StrSplit(output, '\n');
     for (const auto &line : lines)
     {
         std::vector<std::string> fields = absl::StrSplit(line, ':');
         if (fields.size() == 2 && fields[0] == "package")
         {
             std::string package(absl::StripAsciiWhitespace(fields[1]));
-            if (option.debuggable_only)
+            switch (option)
             {
+            case PackageListOptions::kAll: package_list.push_back(package); break;
+            case PackageListOptions::kDebuggableOnly:
                 result = Adb().RunAndGetResult("shell dumpsys package " + package);
                 if (!result.ok())
                 {
                     return result.status();
                 }
                 output = *result;
                 // TODO: find out more reliable way to find if app is debuggable.
+                if (absl::StrContains(output, "DEBUGGABLE"))
+                {
+                    package_list.push_back(package);
+                }
+                break;
+            case PackageListOptions::kNonDebuggableOnly:
+                result = Adb().RunAndGetResult("shell dumpsys package " + package);
+                if (!result.ok())
+                {
+                    return result.status();
+                }
+                output = *result;
                 if (!absl::StrContains(output, "DEBUGGABLE"))
                 {
-                    continue;
+                    package_list.push_back(package);
                 }
+                break;
```

Comment:
also better to add a `default` case and assert in the default case so that in the future if people add a new enum and forget to handle the new one here, the assert would be triggered